server/users.js

```diff
@@ -1568,26 +1571,28 @@ class User extends Chat.MessageContext {
 			this.chatQueue = null;
 		}
 	}
-	isAway() {
-		return this.status && this.status.charAt(0) === '!';
-	}
 	/**
-	 * @param {string} message
+	 * @param {StatusType} type
 	 */
-	setStatus(message) {
-		if (message === this.status) return;
-		this.status = message;
+	setStatusType(type) {
+		if (type === this.statusType) return;
+		this.statusType = type;
 		this.updateIdentity();
 	}
 	/**
 	 * @param {string} message
 	 */
-	setAway(message) {
-		this.setStatus(`!${message}`);
+	setUserMessage(message) {
+		if (message === this.userMessage) return;
+		this.userMessage = message;
+		this.updateIdentity();
 	}
-	clearStatus() {
-		if (!this.status) return;
-		this.status = '';
+	/**
+	 * @param {StatusType} [type]
+	 */
+	clearStatus(type = this.statusType) {
```

Comment:
bit nitpicky but do we need ``setUserMessage('')`` as well as ``clearStatus()``?

Reply:
Good point, we can migrate to just using `clearStatus`.